pkg/gardenadm/translator/static_pod.go

```diff
@@ -0,0 +1,153 @@
+// Secrets that can be injected into an OperatingSystemConfig.
+func Translate(ctx context.Context, c client.Client, o client.Object) ([]extensionsv1alpha1.File, error) {
+	switch obj := o.(type) {
+	case *appsv1.Deployment:
+		return translatePodTemplate(ctx, c, obj.ObjectMeta, obj.Spec.Template)
+	// TODO(rfranzke): Consider adding support for StatefulSet in the future.
+	default:
+		return nil, fmt.Errorf("unsupported object type %T", o)
+	}
+}
+
+func translatePodTemplate(ctx context.Context, c client.Client, objectMeta metav1.ObjectMeta, podTemplate corev1.PodTemplateSpec) ([]extensionsv1alpha1.File, error) {
+	pod := &corev1.Pod{ObjectMeta: podTemplate.ObjectMeta, Spec: podTemplate.Spec}
+	pod.Name = objectMeta.Name
+	pod.Namespace = objectMeta.Namespace
+
+	translateConfig(&pod.Spec)
+	translateContainers(pod.Spec.Containers)
+
+	filesFromVolumes, err := translateVolumes(ctx, c, pod)
+	if err != nil {
+		return nil, fmt.Errorf("failed translating volumes for static pod %s: %w", client.ObjectKeyFromObject(pod), err)
+	}
+
+	staticPodYAML, err := kubernetesutils.Serialize(pod, c.Scheme())
+	if err != nil {
+		return nil, fmt.Errorf("failed serializing static pod manifest for %s to YAML: %w", client.ObjectKeyFromObject(pod), err)
+	}
+
+	return append([]extensionsv1alpha1.File{{
+		Path:        filepath.Join("/", "etc", "kubernetes", "manifests", pod.Name+".yaml"),
+		Permissions: ptr.To[uint32](0644),
+		Content:     extensionsv1alpha1.FileContent{Inline: &extensionsv1alpha1.FileContentInline{Data: staticPodYAML}},
+	}}, filesFromVolumes...), nil
+}
+
+func translateConfig(spec *corev1.PodSpec) {
+	spec.HostNetwork = true
+	spec.PriorityClassName = "system-node-critical"
+}
+
+func translateContainers(containers []corev1.Container) {
+	for i, container := range containers {
+		if container.LivenessProbe != nil && container.LivenessProbe.HTTPGet != nil {
+			containers[i].LivenessProbe.HTTPGet.Host = "127.0.0.1"
+		}
+		if container.ReadinessProbe != nil && container.ReadinessProbe.HTTPGet != nil {
+			containers[i].ReadinessProbe.HTTPGet.Host = "127.0.0.1"
+		}
+		if container.StartupProbe != nil && container.StartupProbe.HTTPGet != nil {
+			containers[i].StartupProbe.HTTPGet.Host = "127.0.0.1"
+		}
+	}
+}
+
+func translateVolumes(ctx context.Context, c client.Client, pod *corev1.Pod) ([]extensionsv1alpha1.File, error) {
+	var (
+		files               []extensionsv1alpha1.File
+		addFileWithHostPath = func(hostPath, fileName, content string, permissions *int32, desiredItems []corev1.KeyToPath) {
+			path := filepath.Join(hostPath, fileName)
+			if len(desiredItems) == 0 || slices.ContainsFunc(desiredItems, func(item corev1.KeyToPath) bool {
+				path = filepath.Join(hostPath, item.Path)
+				return fileName == item.Key
+			}) {
+				files = append(files, extensionsv1alpha1.File{
+					Path:        path,
+					Permissions: ptr.To[uint32](uint32(ptr.Deref(permissions, 0644))), // #nosec G115 -- permissions cannot be negative
```

Comment:
I think that we should use a different default access mode than `0644` considering that some secrets might be stored with these permissions. Maybe `0640` or `0600` work as well. As long as it is not world readable, I am fine.